integration/query_comparison_compat_test.go

```diff
@@ -355,6 +355,9 @@ func testQueryComparisonCompatEq() map[string]queryCompatTestCase {
 		"NoSuchFieldNull": {
 			filter: bson.D{{"no-such-field", bson.D{{"$eq", nil}}}},
 		},
+		"DotNotation": {
+			filter: bson.D{{"v.food.bar", bson.D{{"$eq", "world"}}}},
```

Comment:
I don't see anything in shareddata which matches `v.food.bar` but it's not using `emptyResult` so what is supposed to match this one?

Reply:
Still it should return `emptyResult` if it's incorrect array field. Or I am being blind on what's being matched. After all, both target and compat is matching _something_. So this test proves that implementation matches compat. But I don't what data it matches.